### ISSUE TITLE
All mapgens aside from v6 not generating surface dirt, trees etc

### ISSUE THREAD

**Issue by Lazerbeak12345** [minetest/minetest#11834]
##### Minetest version

RPM version:

```
Minetest 5.4.1 (Linux)
Using Irrlicht 1.8.4
Using LuaJIT 2.1.0-beta3
BUILD_TYPE=Release
RUN_IN_PLACE=0
USE_CURL=1
USE_GETTEXT=1
USE_SOUND=1
USE_FREETYPE=1
STATIC_SHAREDIR="/usr/share/minetest"
STATIC_LOCALEDIR="/usr/share/locale"
```

(Installed via either rpm or Flatpak it has the issue described)

Under flatpak here's the version:
```
Minetest 5.4.1 (Linux)
Using Irrlicht 1.8.4
Using LuaJIT 2.1.0-beta3
BUILD_TYPE=Release
RUN_IN_PLACE=0
USE_CURL=1
USE_GETTEXT=1
USE_SOUND=1
USE_FREETYPE=1
STATIC_SHAREDIR="/app/share/minetest"
STATIC_LOCALEDIR="/app/share/locale"
```

##### OS / Hardware

Operating system: Fedora Linux 35 (Workstation Edition)
CPU: Intel® Core™ i5-2520M CPU @ 2.50GHz × 4

GPU model: N/A issue is with mapgen
OpenGL version: N/A issue is with mapgen

##### Summary

Some mapgens don't generate any biomes? I think. It's only a vast cobblestone plain. 

Here's `debug.txt` when installed with rpm: (note irrlicht was version `1.8.5-1.fc35` in dnf)

```
2021-12-04 19:06:34: WARNING[Main]: Irrlicht: Warning: The library version of the Irrlicht Engine (1.8.5) does not match the version the application was compiled with (1.8.4). This may cause problems.
```

Here's `debug.txt` when installed with flatpak:

```
2021-12-04 19:13:23: ACTION[Main]: World at [/home/nate/.minetest/worlds/a]
2021-12-04 19:13:23: ACTION[Main]: Server for gameid="minetest" listening on 0.0.0.0:63587.
2021-12-04 19:13:25: ACTION[Server]: singleplayer [127.0.0.1] joins game. List of players: singleplayer
2021-12-04 19:13:25: WARNING[Main]: Irrlicht: Could not open file of texture: character.png
```

##### Steps to reproduce

On any game (tried about 10 different games including MineTest Game and WhyNot Game) that supports V7 mapgen, on any map settings (tried all combinations on MTG with V7), with any or no mods added (MTG with no mods can reproduce just fine) do this:

- Create a new world with V7 mapjen.
- Open the world.
- Notice how your character falls down a few blocks and lands on a vast cobblestone plain with interspersed gravel, sand and coal ore.

The mapgens that can repoduce this error:

- V7
- valleys
- carpathian
- v5
- flat
- fractal
- singlenode is completly empty. I don't know if that's normal

That is, _every_ mapgen aside from v6

I've tried setting all settings to default by deleting `minetest.conf`, tried removing all textures, mods and, games. This happens for all user accounts on this computer, has happened since close to version 5.3, and happens if you install via dnf or flatpak.

If you can think of other things from me to test or to show you, the contents or configurations of, feel free to ask. (but yes, I haven't the foggiest, and I'm sure it's something specific about my setup, I just can't figure out what that could be.)

Here's a few screenshots of what it looks like:

![screenshot_20211204_192501](https://user-images.githubusercontent.com/22641188/144731113-d9a6cb98-7032-4777-8d06-36ea1e402734.png)
![screenshot_20211204_192439](https://user-images.githubusercontent.com/22641188/144731115-ae514eb3-5d99-4d20-8c0d-eea9140720e5.png)

**Comment by 0-afflatus** [comment]
As this is completely unreproducible using 5.4.1 installed via apt on Debian, I'd suggest this is at best an edge case.
Has anyone else had problems with the rpm / flatpack package? If so, you need to raise the issue with the package maintainer.
As you are sure it is something specific about your setup, then it's not a bug.
I'd be happy to help you figure out how you broke all your mapgens

**Comment by sfan5** [comment]
This sounds like the issue where some games would inadvertently overwrite the global mapgen settings causing biomes to never work.
You said you deleted minetest.conf and this is still happening but I'm sure that can't be right.
To narrow it down I suggest the following steps:
* (Re)move `~/.minetest`
* Start Minetest with `--verbose`
* Create a new MTG world without mods

If it happens again the entire contents of `debug.txt` and `minetest.conf` (if it exists) are hopefully indicative of where the problem lies.

**Comment by Lazerbeak12345** [comment]
Ok, so moving the `~/.minetest` folder fixed the problem. I'm going to try to narrow down exactly what game or mod caused this: I consider this a security problem. No game or mod should be able to make changes to the world when it isn't applied/running.

**Comment by Lazerbeak12345** [comment]
Ok, after downloading every mod texture pack and game (which I wanted to do anyway for unrelated reasons) I can confirm that without enabling any of them, the bug does _not_ happen. This is a good thing.

**Comment by Lazerbeak12345** [comment]
I've been trying the mods I often use - no luck reproducing. I'm now trying every game.

**Comment by rollerozxa** [comment]
Hades Revisited tends to do this. Check your mapgen flags in the all settings menu and see if the biomes flag is checked. Keep in mind you need to create a new world for it to take effect.

**Comment by Lazerbeak12345** [comment]
Yup. Hades did this. Here's the diff between the current and a former `minetest.conf` (create a new Hades Revisited, create a new Minetest Game with v7):

```
3c3
< mainmenu_last_selected_world = 5
---
> mainmenu_last_selected_world = 8
5c5
< selected_world_path = /home/nate/.minetest/worlds/ab
---
> selected_world_path = /home/nate/.minetest/worlds/a
11c11
< creative_mode = false
---
> creative_mode = true
24,26d23
< mgv6_spflags = nomudflow,nosnowbiomes,notrees,nojungles
< water_level = -5
< mg_flags = nodungeons,notrees,caves,nodecorations,nobiomes
```

**Comment by Lazerbeak12345** [comment]
Isn't this a security problem with minetest itself? I don't think that it should be possible for any game to make the whole of minetest less fun.

**Comment by rollerozxa** [comment]
I believe this has been fixed in 5.5.0-dev already.

**Comment by sfan5** [comment]
Yeah, it has already been fixed by #11352.

also turns out this issue is a duplicate of #11351
